packages/variable-editor/src/components/variables/master/AddVariableDialog.tsx

```diff
@@ -28,14 +36,32 @@ type AddVariableDialogProps = {
 };
 
 export const AddVariableDialog = ({ table, variables, setVariables, setSelectedVariablePath }: AddVariableDialogProps) => {
-  const [selectedNamespace, setSelectedNamespace] = useState('');
+  const [name, setName] = useState('');
+  const [namespace, setNamespace] = useState('');
+
+  const nameValidationMessage = useMemo((): MessageData => {
+    const validationMessage = validateName(name, subRowNamesOfRow(namespace, table));
+    if (!validationMessage) {
+      return;
+    }
+    return { message: validationMessage, variant: 'error' };
+  }, [name, namespace, table]);
+
+  const handleAddVariableDialogOpen = () => {
+    setName(newNodeName(table, 'NewVariable'));
+    setNamespace(keyOfFirstSelectedNonLeafRow(table));
+  };
 
-  const onAddVariableDialogOpen = () => {
-    setSelectedNamespace(keyOfFirstSelectedNonLeafRow(table));
+  const handleNameInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setName(event.target.value);
   };
 
-  const newVariableName = () => {
-    return newNodeName(table, 'NewVariable');
+  const handleNamespaceComboboxChange = (value: string) => {
+    setNamespace(value);
+  };
+
+  const handleNamespaceComboboxInput = (event: FormEvent<HTMLInputElement>) => {
+    setNamespace(event.currentTarget.value);
   };
```

Comment:
why should this function need to know that it will be used for a combobox? it is here to change the namespace...
IMO the first function is anyways completely senseless just call setNamespace on the onChange of the combobox. Not every thing needs to be abstracted into an own function.